api/server/server.go

```diff
@@ -1640,8 +1640,9 @@ func allocateDaemonPort(addr string) error {
 		return fmt.Errorf("failed to lookup %s address in host specification", host)
 	}
 
+	pa := portallocator.Get()
```

Comment:
Then hide the loading somewhere else. This getter pattern doesn't belong in Go code. Arguably, if file io is happening, this shouldn't be a global.